lib/features/home/home_screen.dart

```diff
@@ -52,10 +56,47 @@ class IDHome extends Screen {
             );
           }
 
+          if (state is LocationPermissionNotGrantedState) {
+            WidgetsBinding.instance.addPostFrameCallback((timeStamp) async {
```

Comment:
Instead is using async here, you can send Function callback to your dialog to preform task based on which action user chooses